.travis.yml

```diff
@@ -5,16 +5,51 @@ node_js:
 sudo: false
 cache:
   directories:
+  - docs/vendor/bundle
   - node_modules
 before_install:
 - |
-    git diff --name-only $TRAVIS_COMMIT_RANGE | grep -qvE '(\.md$)|(^(docs|examples))/' || {
+    if [ "$TEST_TYPE" != build_website ] && \
+      git diff --name-only $TRAVIS_COMMIT_RANGE | grep -qvE '(\.md$)|(^(docs|examples))/'
+    then
       echo "Only docs were updated, stopping build process."
       exit
-    }
+    fi
 script:
 - |
-    grunt $TEST_TYPE
+    if [ "$TEST_TYPE" = build_website ]; then
+      if [ "$TRAVIS_BRANCH" = "$REACT_WEBSITE_BRANCH" ] && [ "$TRAVIS_PULL_REQUEST" = false ]; then
+        set -e
+
+        GH_PAGES_DIR="$TRAVIS_BUILD_DIR"/../react-gh-pages
+        echo "machine github.com login reactjs-bot password $GITHUB_TOKEN" >~/.netrc
+
+        # TODO: Can't actually use the Travis cache because shrinkwrapped deps
+        # won't override those already installed. (Without this, the built
+        # JSXTransformer.js uses different dependencies and makes a different
+        # built file.)
+        rm -rf node_modules
+        npm install
+        grunt build  # to create docs/js/{react,JSXTransformer}.js
```

Comment:
I was wondering about this step. Would it maybe make sense to just exclude these always? Then we intentionally check the right ones in when we ship a new release?